timur/lib/client/jsx/components/query/query_filter_control.tsx

```diff
@@ -1,219 +1,183 @@
-    graph
-  });
-
-  const filterOperator = useMemo(() => {
-    return new FilterOperator(attributeType, filter.operator, isColumnFilter);
-  }, [attributeType, filter.operator, isColumnFilter]);
-
-  useEffect(() => {
-    // When user selects a different attribute, update the type
-    patchFilter({
-      ...filter,
-      attributeType
-    });
-  }, [filter.attributeName, attributeType]);
-
   const handleModelSelect = useCallback(
     (modelName: string) => {
       patchFilter({
         modelName,
-        attributeName: '',
-        operator: '',
-        operand: '',
-        attributeType: ''
+        anyMap: {},
+        clauses: [
+          {
+            ...EmptyQueryClause
+          }
+        ]
       });
     },
     [patchFilter]
   );
 
-  const handleAttributeSelect = useCallback(
-    (attributeName: string) => {
+  const handlePatchClause = useCallback(
+    (clause: QueryClause, index: number) => {
+      let updatedClauses = [...filter.clauses];
+      updatedClauses[index] = clause;
       patchFilter({
         ...filter,
-        attributeName,
-        operator: '',
-        operand: ''
+        clauses: updatedClauses
       });
     },
-    [filter, patchFilter]
+    [patchFilter, filter]
   );
 
-  const handleOperatorSelect = useCallback(
-    (operator: string) =>
+  const handleRemoveClause = useCallback(
+    (index: number) => {
+      let updatedClauses = [...filter.clauses];
+      updatedClauses.splice(index, 1);
       patchFilter({
         ...filter,
-        operator: filterOperator.magmify(operator)
-      }),
-    [filter, patchFilter, filterOperator]
-  );
-
-  const handleOperandChange = useCallback(
-    (operand: string) => {
-      patchFilter({
-        ...filter,
-        operand: filterOperator.formatOperand(operand)
+        clauses: updatedClauses
       });
     },
-    [patchFilter, filter, filterOperator]
+    [patchFilter, filter]
   );
 
-  const handleOperandChangeWithDebounce = useCallback(
-    (value: string) => {
-      debouncer.ready(() => handleOperandChange(value));
-      setOperandValue(value);
+  const handleAddClause = useCallback(() => {
+    patchFilter({
+      ...filter,
+      clauses: [...filter.clauses].concat([
+        {...EmptyQueryClause, modelName: filter.modelName}
+      ])
+    });
+  }, [patchFilter, filter]);
+
+  const handleClauseAnySelect = useCallback(
+    (val: string, clause: QueryClause, index: number) => {
+      handlePatchClause(
+        {
+          ...clause,
+          any: val === 'Any'
+        },
+        index
+      );
     },
-    [handleOperandChange, debouncer]
+    [handlePatchClause]
   );
 
-  // When the operand value changes, follow it
-  useEffect(() => {
-    if (filter.operand !== previousOperandValue) {
-      debouncer.reset();
-      setOperandValue(filter.operand);
-      setPreviousOperandValue(filter.operand);
-    }
-  }, [filter.operand, debouncer, previousOperandValue]);
+  const modelChildren = useMemo(() => {
```

Comment:
@graft , I made an assumption here that clauses can only be appending to either the selected filter's model, or it's children models. Because if you go deeper (or elsewhere in the graph), I'm not sure how you would handle the any / every selection at each step in the UI, and it wasn't clear from your use case if that would ever be desireable versus just adding filters at the top level?

I guess one alternative is to allow users to just go "down" the graph, like with slices (i.e. timepoint -> immunoassay -> analyte in COMET), which would be like 2 any / every selectors but without the ability to add slices for each model (nesting all these would get confusing, I think). We would also have to check that the API itself would construct the sub-selects correctly for deeper filters.